aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/AWSCognitoAuthSignUpOptions.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.options;
+
+import androidx.annotation.NonNull;
+import androidx.core.util.ObjectsCompat;
+
+import com.amplifyframework.auth.AuthUserAttribute;
+import com.amplifyframework.auth.options.AuthSignUpOptions;
+import com.amplifyframework.util.Immutable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public final class AWSCognitoAuthSignUpOptions extends AuthSignUpOptions {
+    private final Map<String, String> validationData;
+
+    /**
+     * Advanced options for signing in.
+     * @param userAttributes Additional user attributes which should be associated with this user on registration
+     * @param validationData A map of custom key/values to be sent as part of the sign up process
+     */
+    protected AWSCognitoAuthSignUpOptions(
+            List<AuthUserAttribute> userAttributes,
+            Map<String, String> validationData
+    ) {
+        super(userAttributes);
+        this.validationData = validationData;
+    }
+
+    /**
+     * Get a map of custom key/values to be sent as part of the sign up process.
+     * @return a map of custom key/values to be sent as part of the sign up process
+     */
+    @NonNull
+    public Map<String, String> getValidationData() {
+        return validationData;
+    }
+
+    @NonNull
+    public static CognitoBuilder builder() {
+        return new CognitoBuilder();
+    }
+
+    @Override
+    public int hashCode() {
+        return ObjectsCompat.hash(
+                getUserAttributes(),
+                getValidationData()
+        );
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        } else if (obj == null || getClass() != obj.getClass()) {
```

Comment:
If I have empty Cognito sign-up options, and empty Third party sign-up options, are the options the same?

Reply:
Hmm, I guessss so... will update

Reply:
In attempting to implement this here, I don't think this is a helpful thing to check and opens a whole new can of worms where I don't know what all other classes implement the base class so I can't check whether they have options defined which I don't have defined and thus are not equal. 

I'm going to stick with the stricter (and I think more traditional) definition that for two objects to be equal, they have to at least be the same class.